lib/job-iteration/iteration.rb

```diff
@@ -262,7 +289,7 @@ def output_interrupt_summary
     end
 
     def job_should_exit?
-      if ::JobIteration.max_job_runtime && start_time && (Time.now.utc - start_time) > ::JobIteration.max_job_runtime
+      if job_iteration_max_job_runtime && start_time && (Time.now.utc - start_time) > job_iteration_max_job_runtime
```

Comment:
Sorry to comment on such an old PR, please let me know if you'd rather I convert this to an issue, or move the conversation somewhere else.

This change actually creates a subtle behaviour change.  Previously, we read the global configuration at job runtime.  Now, we read the global configuration when `JobIteration::Iteration` is loaded, subsequent changes are ignored.

Today, we noticed that maintenance tasks were never pausing because that gem were setting a default job runtime in an `after_initialze` block that was run after this module was included.

I'm not actually sure which option is better.  Just wanted to point out the difference in behaviours.  In the meantime, I've proposed Shopify/maintenance_tasks#918 to fix maintenance tasks under the assumption that the current behaviour of job-iteration is correct.

Reply:
I can try to have a look on Monday, but I think the fix would be to override the method defined by `class_attribute` to delegate to the top level default, as opposed to setting it to the default value at the time the method was defined.

Reply:
#436 should fix this.